src/Uno.UI/UI/Xaml/Controls/Icon/IconElement.macOS.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Windows.UI.Xaml.Controls;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using AppKit;
+
+namespace Windows.UI.Xaml.Controls
+{
+	public partial class IconElement
+    {
+		partial void UnregisterSubView()
+		{
+			if (Subviews.Length > 0)
+			{
+				Subviews[0].RemoveFromSuperview();
+			}
+		}
+
+		partial void RegisterSubView(NSView child)
+		{
+			if (Subviews.Length != 0)
+			{
+				throw new Exception("A Xaml IconElement may not contain more than one child.");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4937867)